app/models/genre.rb

```diff
@@ -0,0 +1,12 @@
+class Genre < ApplicationRecord
+  has_and_belongs_to_many :books
+
+  validates :name, presence: true
+
+  scope :ordered, -> { order(name: :asc) }
+
+  scope :search, -> (search_term) {
+    search_term_with_wildcards = "%#{search_term}%"
+    where("name LIKE ? OR name LIKE ?",
+    search_term_with_wildcards, search_term_with_wildcards) }
```

Comment:
When we have blocks that span multiple lines it is customary in Ruby to use the `do` `end` syntax

```ruby
  scope :search, -> (search_term) do
    search_term_with_wildcards = "%#{search_term}%"
    where("name LIKE ? OR name LIKE ?", search_term_with_wildcards, search_term_with_wildcards)
  end
```